cmd/edit/cluster/cmd.go

```diff
@@ -189,10 +216,185 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	enableProxy := false
+	useExistingVPC := false
+	var httpProxy *string
+	var httpProxyValue string
+	if cmd.Flags().Changed("http-proxy") {
+		httpProxyValue = args.httpProxy
+		httpProxy = &httpProxyValue
+	}
+	var httpsProxy *string
+	var httpsProxyValue string
+	if cmd.Flags().Changed("https-proxy") {
+		httpsProxyValue = args.httpsProxy
+		httpsProxy = &httpsProxyValue
+	}
+	var additionalTrustBundleFile *string
+	var additionalTrustBundleFileValue string
+	if cmd.Flags().Changed("additional-trust-bundle-file") {
+		additionalTrustBundleFileValue = args.additionalTrustBundleFile
+		additionalTrustBundleFile = &additionalTrustBundleFileValue
+	}
+
+	if httpProxy != nil || httpsProxy != nil || additionalTrustBundleFile != nil {
+		enableProxy = true
+		useExistingVPC = true
+	}
+
+	if len(cluster.AWS().SubnetIDs()) > 0 {
+		useExistingVPC = true
+	}
```

Comment:
This is likely wrong. Since cluster-wide proxy only applies to BYO-VPC clusters, if the user sets any of the proxy flags, but there are no subnets in the cluster, then the command should error out.

Reply:
@vkareh 
```
if len(cluster.AWS().SubnetIDs()) > 0 {
	useExistingVPC = true
}

if useExistingVPC && !enableProxy && interactive.Enabled() {
	enableProxyValue, err := interactive.GetBool(interactive.Input{
		Question: "Update cluster-wide proxy",  **<---- I want to present this question only if I know that this is BYOVPC cluster which is indicated by the first 'if' check above**.
```

Reply:
Correct, but my point is that we should go further than this and have something like
```go
if len(cluster.AWS().SubnetIDs()) > 0 && (httpProxy != "" || httpsProxy != "" || additionalTrustBundle != "") {
  reporter.Errorf("Cluster-wide proxy is not supported on clusters using the default VPC")
  os.Exit(1)
}
```